cli/commandhandler.h

```diff
@@ -22,9 +24,11 @@ Transaction createTransaction(ih::wrapper::TransactionInputWrapper const &txWrap
              txWrapper.getOptions());
 }
 
-void signTransaction(Transaction &transaction, bytes const& seed)
+// TODO: replace keyFile param with template param once
```

Comment:
Well, not necessarily a type parameter, right? Why not receive an interface / abstract class (with several implementations), such as `ISecretKeyProvider`?

Reply:
Good point.
I can implement a `ISecretKeyProvider` interface and make `PemFileReader` and `KeyFileReader` derive from it.